### PR TITLE
Issue 46 update readme on release

### PR DESCRIPTION
**Pull request by skyerus**

### CODE REVIEW

**Comment by skyerus** [comment]
https://github.com/open-feature/open-feature-operator/issues/46

**Comment by skyerus** [comment]
Tested on my forked repo that this behaves as expected

File: .github/workflows/release-workflow.yaml [hunk 1]
```diff
@@ -1,15 +1,9 @@
-name: release
+name: release-workflow
```
Comment (skyerus) [inline .github/workflows/release-workflow.yaml:1]: separated into a separate workflow as we should only update the readme versions on a tag push

File: .github/workflows/release-tag.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      - "v*.*.*"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
```
Comment (beeme1mr) [inline .github/workflows/release-tag.yaml:15]: Do we need this anymore? I don't think we're using the schema submodule anymore. @james-milligan, can you confirm?

File: .github/workflows/release-tag.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          find: "https://github\\.com/open-feature/open-feature-operator/releases/download/v[\\d]+\\.[\\d]+\\.[\\d]+"
+          replace: "https://github.com/open-feature/open-feature-operator/releases/download/${{ steps.get_version.outputs.VERSION }}"
+          regex: true
+          include: README.md
+
+      - name: Commit files
+        run: |
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          git commit -s -m "Updated README.md version" -a
```
Comment (beeme1mr) [inline .github/workflows/release-tag.yaml:42]:
```suggestion
          git commit -s -m "docs: update README.md version" -a
```

Updating to include a conventional commit message. We don't enforce this on the repo yet but we may in the future.

File: .github/workflows/release-workflow.yaml [hunk 4]
```diff
@@ -18,11 +12,11 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v2
         with:
-         submodules: recursive
+          submodules: recursive
```
Comment (beeme1mr) [inline .github/workflows/release-workflow.yaml:15]: Same question as above. Do we still need this?

Comment (skyerus) [reply .github/workflows/release-workflow.yaml]: Removed

File: .github/workflows/release-tag.yaml (hunk 2 above)
Comment (skyerus) [reply .github/workflows/release-tag.yaml]: Removed

**Review by beeme1mr — APPROVED** [review]

**Review by thschue — APPROVED** [review]
lgtm

**Review by AlexsJones — APPROVED** [review]

**Pull request merged**